### PR TITLE
refactor(searcher): Change SearcherWithRandomSeed to StochasticSearcher

### PR DESCRIPTION
**Pull request by aaronkl**
*Description of changes:*

Changes SearcherWithRandomSeed to StochasticSearcher to highlight that any randomized algorithm - algorithms that make randomized decisions internally - should derive from this class. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: syne_tune/optimizer/schedulers/searchers/__init__.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
     impute_points_to_evaluate,
 )
 from syne_tune.optimizer.schedulers.searchers.searcher_base import (  # noqa: F401
-    SearcherWithRandomSeed,
+    StochasticSearcher,
     SearcherWithRandomSeedAndFilterDuplicates,
```
Comment (mseeger) [inline syne_tune/optimizer/schedulers/searchers/__init__.py:23]: Could also rename this to `StochasticAndFilterDuplicatesSearcher`

**Review by mseeger — CHANGES_REQUESTED** [review]
Could you also please grep all the docs and do the replacement there as well?

**Review by mseeger — APPROVED** [review]
But let us merge this only once the new release is out, to be safe

**Comment by wesk** [comment]
Also, would you please update the title of this PR to follow the conventionalcommits convention guidelines as described here: https://github.com/awslabs/syne-tune/blob/main/CONTRIBUTING.md#contributing-via-pull-requests

And also yes, please wait to merge until after this release

**Pull request merged**